cmake/utils.cmake

```diff
@@ -145,3 +145,15 @@ function(list_add_prefix
     endforeach()
     set(${list_out} ${result} PARENT_SCOPE)
 endfunction()
+
+# string() has JOIN option but only since 3.12.
+function(string_join
+    glue
+    string_out
+)
+    set(result "")
+    foreach(item IN LISTS ARGN)
+        string(CONCAT result "${i}${glue}")
+    endforeach()
+    set(${list_out} ${result} PARENT_SCOPE)
+endfunction()
```

Comment:
This function simply doesn't work...

Reply:
Sorry, I relied on building with `-DENABLE_UB_SANITIZER`, and it didn't give any errors, so I didn't double-check.

Fixed.